### PR TITLE
[GEOS-10856] geoserver monitor plugin - scaling troubles

### PR DESCRIPTION
**Pull request by hyperman2**
[GEOS-10856](https://osgeo-org.atlassian.net/browse/GEOS-10856)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [ ] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: doc/en/user/source/extensions/monitoring/configuration.rst [hunk 1]
```diff
@@ -136,6 +136,18 @@ is added::
 
    /wfs
 
+Monitoring threads
+------------------
+You can choose the number of post processor threads by configuring the ``postprocessorthreads``
```
Comment (aaime) [inline doc/en/user/source/extensions/monitoring/configuration.rst:141]:
```suggestion
You can choose the number of post processor threads by configuring the ``postProcessorThreads``
```

The file is using camelCase for all other properties.

File: doc/en/user/source/extensions/monitoring/configuration.rst [hunk 2]
```diff
@@ -136,6 +136,18 @@ is added::
...
+property in the ``monitor.properties`` file.  The default is 2.
+
+DNS cache configuration
+-----------------------
+The reverseDNS post processor caches its result.  You can modify the cache configuration
+by configuring the ``dnscacheconfiguration`` property in the ``monitor.properties`` file.
```
Comment (aaime) [inline doc/en/user/source/extensions/monitoring/configuration.rst:147]: Same as above

```suggestion
by configuring the ``dnsCacheConfiguration`` property in the ``monitor.properties`` file.
```

**Review by aaime — CHANGES_REQUESTED** [review]
Looks good in general, see comments below.

File: src/extension/monitor/core/src/main/java/org/geoserver/monitor/MonitorConfig.java [hunk 3]
```diff
@@ -65,6 +65,10 @@ public static enum BboxMode {
     Exception error;
     private GeoServerResourceLoader loader;
 
+    static final int POSTPROCES_THREADS_DEFAULT = 2;
+
+    static final String DNS_CACHE_DEFAULT = "expireAfterWrite=15m";
```
[the remark this answers is not in the document]
Comment (hyperman2) [reply src/extension/monitor/core/src/main/java/org/geoserver/monitor/MonitorConfig.java:70]: Thanks, I'll look at this in the evening

Regarding DNS_CACHE_DEFAULT: I see a few options:  
1) Change to "expireAfterWrite=15m,softValues" should provide the same behaviour as soft references.
2) Change to "expireAfterWrite=15m,maximumSize=100" should provide the same maximum as in the current code.  

As the current code only reads the cache, never writes it, the current cache is always empty, so I can't look at what happens right now.    I presume soft refs are more heavy on the GC as they need special handling, and the cost of a String is very low, so I propose option 2, maybe even raising the maximum to 1000.

**Comment by hyperman2** [comment]
Sorry, didn't see the 'Request re-review button'.  All remarks have been adapted

**Review by aaime — APPROVED** [review]

**Comment by hyperman2** [comment]
Thank you very much

**Pull request merged**